ablySpec/RestClient.swift

```diff
@@ -45,6 +45,32 @@ class RestClient: QuickSpec {
                     expect(publishTask.error?.code).toEventually(equal(40005))
                 }
 
+                it("should accept a token") {
+                    let options = AblyTests.commonAppSetup()
+                    let client = ARTRest(key: options.key!)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.auth.requestToken(ARTAuthTokenParams(), withOptions: nil) { tokenDetails, error in
+                            if let e = error {
+                                XCTFail(e.description)
+                                done()
+                                return
+                            }
+
+                            guard let tokenDetails = tokenDetails else {
+                                XCTFail("expected tokenDetails not to be nil when error is nil")
+                                done()
+                                return
+                            }
+
+                            let client = ARTRest(token: tokenDetails.token)
```

Comment:
In other languages we have an overloaded constructor so that you can pass in just a token or API key string, or options.  If this is not possible in iOS then it's absolutely fine, but if it is possible it's a nice touch.